### ISSUE TITLE
Tell crawlers to stop indexing rust docs

### ISSUE THREAD

**Issue by stevenengler** [shadow/shadow#3443]
We publish the rustdoc output to https://shadow.github.io/docs/rust in https://github.com/shadow/docs. This unfortunately gets picked up by the google indexer, so searches like "rust neli" return results for our documentation pages.

I'm not sure that publishing the rust docs are even useful anymore, but if we keep them we should also use the rustdoc [`--html-in-header`](https://doc.rust-lang.org/rustdoc/command-line-arguments.html#--html-in-header-include-more-html-in-head) flag to add a `<meta name="robots" content="noindex">` or similar to each page.

**Comment by stevenengler** [comment]
@robgjansen @sporksmith I plan to remove the rust docs from https://shadow.github.io/docs/rust sometime in the next few weeks. If either of you still find them useful, let me know and I'll leave them.

**Comment by sporksmith** [comment]
No objection here

**Comment by robgjansen** [comment]
OK. I think the rust docs are probably more useful for people less familiar with Shadow to help them learn about the project structure etc, but also many of our functions and objects are not well documented so the docs are not living up to their full potential at this time.

I think we can still generate them locally with something like `(cd src && cargo doc --workspace --exclude shadow-tests)`, which is currently documented in [the guide](https://shadow.github.io/docs/guide/coding.html?highlight=mdbook#building-the-rust-docs).

After you remove them, we'll need to remove the link to https://shadow.github.io/docs/rust from the https://shadow.github.io front page

**Comment by stevenengler** [comment]
Yeah, the docs aren't the greatest, plus it's not really useful for linking to types/methods since the website isn't versioned, so links will go stale as we move code around. It's usually better to just link to the source code on github at a specific commit hash.

We can still generate the rust docs locally. And we can re-add it back to https://shadow.github.io/docs/rust easily later if we think it would be useful.

My thinking is that I don't know of any good use cases that we should keep it around for, and it's something that breaks every once in a while and needs to be maintained. And since the rust doc output isn't stable, it causes our shadow/docs repo to be huge after months/years of daily rebuilds of the static site. So I think it's simplest just to get rid of it. But if anyone would prefer to keep it around, or is just leaning towards keeping it, that's okay too.

**Comment by sporksmith** [comment]
Yeah, IMO the main use-case for rust docs is for crates that you're using, but aren't part of the project you're working on. For "the project you're currently working on" IMO it's usually easier to just read the source and source-comments directly, especially with decent IDE support. And since we explicitly don't support using any of our crates outside of our shadow I'm not sure who that leaves as a target audience.
